### PR TITLE
add Channels, EventSources, OperatorBacked, HelmCharts menu action to Add To Project context menu

### PR DESCRIPTION
**Pull request by sahil143**
**Fixes**: https://issues.redhat.com/browse/ODC-4914

**Analysis / Root cause**: missing Channels, EventSources, OperatorBacked, HelmCharts from Add to Project sub menu

**Solution Description**: added Channels, EventSources, OperatorBacked, HelmCharts menu action

**Screen shots / Gifs for design review**: 
Screenshot 2020-10-05 at 9 35 11 PM
image

**Unit test coverage report**: 

**Test setup:**

**Browser conformance**: 

- [X] Chrome
- [ ] Firefox
- [ ] Safari
- [ ] Edge

### CODE REVIEW

**Comment by andrewballantyne** [comment]
/cc @rohitkrai03 @christianvogt 

I'm seeing a lack of extensibility here, and Helm stuff showing up in shared. Need Rohit / Christian to chime in here with suggestions.

**Comment by christianvogt** [comment]
I'd like to keep helm out of shared. It will become it's own package soon.
Until we improve the actions extension 4.7, I'm ok with the hard coding as it's already done for the other actions on the graph.

**Comment by invincibleJai** [comment]
@sahil143 one issue i see but this is not with this PR however as we have more options so it's easily reproduced , if we try to add to project from lower quadrant of screen, we may need to track this as separate issue

Oct-07-2020 17-40-00

**Comment by bgliwa01** [comment]
will there be a separate PR for adding Operator Backed to an existing component?

**Comment by sahil143** [comment]
>will there be a separate PR for adding Operator Backed to an existing component?

@bgliwa01 Do you mean through connector? If yes, that will be another story. This PR only takes care of adding action items to the context menu.

**Comment by sahil143** [comment]
> @sahil143 one issue i see but this is not with this PR however as we have more options now so it's easily reproduced, if we try to add to project from lower quadrant of screen, we may need to track this as separate issue


@invincibleJai I could reproduce it on the `patternfly-react` repo `react-topology` package itself.. So it will be fixed there. I have raised and issue to track this https://issues.redhat.com/browse/ODC-4988

react-topology demo app screenshot
Screenshot 2020-10-08 at 9 51 48 PM

**Review by bgliwa01 — APPROVED** [review]
lgtm

File: frontend/packages/dev-console/src/components/helm/HelmChartsIcons.tsx [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+type HelmChartsIconProps = {
+  className?: string;
+  style?: React.CSSProperties;
+};
+
+const HelmChartsIcon: React.FC<HelmChartsIconProps> = ({ className, style }) => (
+  <img className={className} style={style} src={helmIcon} alt="Helm Charts Logo" />
+);
+
+export default HelmChartsIcon;
```
Comment (invincibleJai) [inline frontend/packages/dev-console/src/components/helm/HelmChartsIcons.tsx:13]: nit: shall we have file name as well `HelmChartsIcon`

File: frontend/packages/knative-plugin/src/actions/add-channel.tsx [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+import { ImportOptions } from '@console/dev-console/src/components/import/import-types';
+import * as channelIcon from '../imgs/channel.svg';
+
+const eventChannelStyles = {
+  width: '1em',
+  height: '1em',
+};
+const EventChannelIcon: React.FC = () => {
+  return <img style={eventChannelStyles} src={channelIcon} alt="" />;
+};
```
Comment (invincibleJai) [inline frontend/packages/knative-plugin/src/actions/add-channel.tsx:15]: nit: can use implicit return here

**Review by karthikjeeyar — APPROVED** [review]
Verified locally, works fine
/lgtm

**Comment by invincibleJai** [comment]
/approve

Verified, works as expected.

**Pull request merged**